packages-exp/auth-exp/test/integration/flows/anonymous.test.ts

```diff
@@ -0,0 +1,121 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import {
+  createUserWithEmailAndPassword,
+  EmailAuthProvider,
+  linkWithCredential,
+  signInAnonymously,
+  signInWithEmailAndPassword,
+  updateEmail,
+  updatePassword
+} from '@firebase/auth-exp/index.browser';
+import { OperationType } from '@firebase/auth-types-exp';
+import { FirebaseError } from '@firebase/util';
+
+import {
+  initIntegrationTestContext,
+  randomEmail
+} from '../../helpers/integration/helpers';
+
+use(chaiAsPromised);
+
+describe('Integration test: anonymous auth', () => {
+  const auth = initIntegrationTestContext();
+
+  it('signs in anonymously', async () => {
+    const userCred = await signInAnonymously(auth);
+    expect(auth.currentUser).to.eq(userCred.user);
+    expect(userCred.operationType).to.eq(OperationType.SIGN_IN);
+
+    const user = userCred.user;
+    expect(user.isAnonymous).to.be.true;
+    expect(typeof user.uid).to.eq('string');
+  });
+
+  it('second sign in on the same device yields same user', async () => {
+    const { user: userA } = await signInAnonymously(auth);
+    const { user: userB } = await signInAnonymously(auth);
+
+    expect(userA.uid).to.eq(userB.uid);
+  });
+
+  context('email/password interaction', () => {
+    let email: string;
+
+    beforeEach(() => {
+      email = randomEmail();
+    });
+
+    it('anonymous / email-password accounts remain independent', async () => {
+      let anonCred = await signInAnonymously(auth);
+      const emailCred = await createUserWithEmailAndPassword(
+        auth,
+        email,
+        'password'
+      );
+      expect(emailCred.user.uid).not.to.eql(anonCred.user.uid);
+
+      await auth.signOut();
+      anonCred = await signInAnonymously(auth);
+      const emailSignIn = await signInWithEmailAndPassword(
+        auth,
+        email,
+        'password'
+      );
+      expect(emailCred.user.uid).to.eql(emailSignIn.user.uid);
+      expect(emailSignIn.user.uid).not.to.eql(anonCred.user.uid);
+    });
+
+    it('account can be upgraded by setting email and password', async () => {
+      const { user } = await signInAnonymously(auth);
+      await updateEmail(user, email);
+      await updatePassword(user, 'password');
+
+      const anonId = user.uid;
+      await auth.signOut();
+      expect(
+        (await signInWithEmailAndPassword(auth, email, 'password')).user.uid
+      ).to.eq(anonId);
+    });
+
+    it('account can be linked using email and password', async () => {
+      const { user } = await signInAnonymously(auth);
```

Comment:
I'd suggest renaming the variable to `anonUser` (like below) to make the test more readable. You can also drop the `id` constant -- it's not much longer but clearer to read in its full form.

Reply:
Done